frontend/src/components/profile/UserForm.tsx

```diff
@@ -14,30 +14,85 @@ type UserFormProps = {
     editable: boolean
     marginTop: number
     showEditButton: boolean
-    onEditButtonClick?(): void
+}
+
+interface IInputFields {
```

Comment:
Ich würde dir das I vor dem Interface nicht empfehlen.
Soweit ich weiß, gilt das mittlerweile wohl eher als veraltet.
Ein Grund ist zB die Lesbarkeit. Typischerweise würde man eher `InterfaceClass` und `InterfaceClassImpl` schreiben.
Das gilt aber auch eher für Java.

In diesem Fall nutzt du das interface Keyword ja als Typdeklaration. Falls ich mit der Annahme richtig liege, würde die Interface Semantik ja eh weniger passen und dann würde ich auch einfch `type` schreiben.
Wie im Kommentar drüber erwähnt, sind interface und type aber natürlich recht austauschbar.
In dem Fall, entscheid dich doch gerne für eins und zieh es innerhalb der gesamten Codebase durch.

Korrigier mich gerne, wenn es caluclated ist warum du vorhe type und jetzt interface benutzt :)